htdocs/plugins/preloads/googleanalytics.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * googleanalytics ga4.php
+ * Created by david on 13/02/2022 22:41
+ *
+ */
+
+/**
+ * preload class to include the Google Analytics V4 tag in the Head of the webpage, after replacing the smarty variable with the GA4 property code
+ * This can certainly be improved by checking whether we can, based on the preferences by our users (GDPR and cookies)
+ */
+class IcmsPreloadGoogleanalytics extends icms_preload_Item {
+
+    function eventBeforeFooter()
+    {
+        global $xoopsTpl;
+        global $icmsTheme;
+        global $icmsConfigMetaFooter;
+
+
+        try {
+
+            if ($icmsConfigMetaFooter['use_google_analytics'] && isset($icmsConfigMetaFooter['google_analytics']) && $icmsConfigMetaFooter['google_analytics'] != '') {
+				if(substr( $icmsConfigMetaFooter['use_google_analytics'], 0, 2 ) === "G-")
+				{
+					$this->insertGA4Tag($icmsConfigMetaFooter['use_google_analytics']);
+				}
+				else
+				{
+					$this->insertUniversalAnalyticsTag($icmsConfigMetaFooter['use_google_analytics']);
+				}
+            }
+            else {
+                echo 'error getting icmsConfigMetaFooter';
+            }
```

Comment:
Where will this display? The site may not have GA enabled or configured intentionally